### PR TITLE
chore: define rust-tools.txt for rust dev tools

### PR DESCRIPTION
**Pull request by everpcpc**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

* fix dev setup install precommand, an old bug exposed by `-e`
* define rust-tools.txt for rust dev tools

## Changelog

- Build/Testing/CI

## Related Issues

for: https://github.com/datafuselabs/databend/issues/4595

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by BohuTANG** [comment]
@mergify update

**Comment by everpcpc** [comment]
code seems broken on new rust toolchain

**Comment by BohuTANG** [comment]
@mergify update

**Review by ZhiHanZ — APPROVED** [review]

File: .github/workflows/platform.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@ on:
       - main
     paths:
       - "rust-toolchain.toml"
+      - "rust-tools.txt"
```
Comment (BohuTANG) [inline .github/workflows/platform.yml:9]: Move the rust-tools.txt to scripts/setup/?
So we can keep the root directory cleaned.

**Comment by BohuTANG** [comment]
```
Dockerfile:19
--------------------
  18 |     COPY scripts/setup/dev_setup.sh /build/scripts/setup/dev_setup.sh
  19 | >>> RUN apk --update add --no-cache bash && \
  20 | >>>     chmod +x /build/scripts/setup/dev_setup.sh && \
  21 | >>>     /build/scripts/setup/dev_setup.sh && \
  22 | >>>     rm -rf /root/.cache/* && \
  23 | >>>     rm -rf /root/.cargo/registry/*
  24 |     
--------------------
error: failed to solve: process "/bin/sh -c apk --update add --no-cache bash &&     chmod +x /build/scripts/setup/dev_setup.sh &&     /build/scripts/setup/dev_setup.sh &&     rm -rf /root/.cache/* &&     rm -rf /root/.cargo/registry/*" did not complete successfully: exit code: 127
Error: buildx failed with: error: failed to solve: process "/bin/sh -c apk --update add --no-cache bash &&     chmod +x /build/scripts/setup/dev_setup.sh &&     /build/scripts/setup/dev_setup.sh &&     rm -rf /root/.cache/* &&     rm -rf /root/.cargo/registry/*" did not complete successfully: exit code: 127
```

**Pull request merged**